src/parser.rs

```diff
@@ -415,7 +420,8 @@ rule scenario() -> Scenario
 rule tag_char() -> &'input str
     = s:$([_]) {?
         let x = s.chars().next().unwrap();
-        if x.is_alphanumeric() || x == '_' || x == '-' {
+        // `)` isn't allowed, as it would collide with TagExpression.
+        if !x.is_whitespace() && !"\r\n@)".contains(x) {
```

Comment:
`\r`, `\n` and even `\t` are [covered by `.is_whitespace()`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=fn%20main()%20%7B%0A%20%20%20%20assert!(%27%5Ct%27.is_whitespace())%3B%0A%20%20%20%20assert!(%27%5Cr%27.is_whitespace())%3B%0A%20%20%20%20assert!(%27%5Cn%27.is_whitespace())%3B%0A%7D) already.